### ISSUE TITLE
SQL Error: 1032 (SQLITE_READONLY_DBMOVED) while syncing

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#7276]
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/1abea72f-4314-499d-b519-9827722b2da1

https://www.sqlite.org/rescode.html#readonly_dbmoved

https://www.reddit.com/r/medicalschoolanki/comments/iyv9g2/after_i_have_updated_to_anking_v8_from_orginal/

```
	
0	--------- beginning of main
1	09-23 23:33:13.409 I/AnkiDroid( 1694): Closing collection for full sync
2	09-23 23:33:13.519 I/AnkiDroid( 1694): Closing Collection Log
3	09-23 23:33:13.519 I/AnkiDroid( 1694): Collection closed
4	09-23 23:33:19.238 I/AnkiDroid( 1694): AnkiActivity::onPause - DeckPicker
5	09-23 23:33:20.585 I/AnkiDroid( 1694): AnkiActivity::onStop - DeckPicker
6	09-23 23:33:52.033 I/AnkiDroid( 1694): AnkiActivity::onStart - DeckPicker
7	09-23 23:33:52.036 I/AnkiDroid( 1694): AnkiActivity::onResume - DeckPicker
8	09-23 23:33:52.045 I/AnkiDroid( 1694): Begin openCollection: /sdcard/AnkiDroid/collection.anki2
9	09-23 23:33:52.053 I/AnkiDroid( 1694): Opening Collection Log
10	09-23 23:33:52.187 I/AnkiDroid( 1694): End openCollection: /sdcard/AnkiDroid/collection.anki2
11	09-23 23:37:21.929 I/AnkiDroid( 1694): Full Sync Success: Overwritten collection with downloaded file
12	09-23 23:37:21.929 I/AnkiDroid( 1694): Reopening Database
13	09-23 23:37:21.933 I/AnkiDroid( 1694): Opening Collection Log
14	09-23 23:37:21.953 I/AnkiDroid( 1694): Sync - Performing media sync
15	09-23 23:37:23.008 I/AnkiDroid( 1694): Sync Finished - Closing Collection
16	09-23 23:37:23.011 I/AnkiDroid( 1694): Closing Collection Log
17	09-23 23:37:23.012 I/AnkiDroid( 1694): Collection closed
18	09-23 23:37:23.058 I/AnkiDroid( 1694): Sync was successful
19	09-23 23:37:23.058 I/AnkiDroid( 1694): Full Download Completed
20	09-23 23:37:24.025 I/AnkiDroid( 1694): Updating deck list UI
21	09-23 23:38:10.118 I/AnkiDroid( 1694): DeckPicker:: Sync button pressed
22	09-23 23:38:10.177 I/AnkiDroid( 1694): Sync - starting sync
23	09-23 23:38:10.252 E/SQLiteLog( 1694): (1032) statement aborts at 22: [UPDATE col SET decks=?,dconf=?] attempt to write a readonly database
24	09-23 23:38:10.270 E/AnkiDroid( 1694): Connection/ doInBackgroundSync error
25	09-23 23:38:10.270 E/AnkiDroid( 1694): android.database.sqlite.SQLiteReadOnlyDatabaseException: attempt to write a readonly database (code 1032)
26	09-23 23:38:10.270 E/AnkiDroid( 1694): #################################################################
27	09-23 23:38:10.270 E/AnkiDroid( 1694): Error Code : 1032 (SQLITE_READONLY_DBMOVED)
28	09-23 23:38:10.270 E/AnkiDroid( 1694): Caused By : Database or Journal file have been removed.
29	09-23 23:38:10.270 E/AnkiDroid( 1694): (attempt to write a readonly database (code 1032))
30	09-23 23:38:10.270 E/AnkiDroid( 1694): #################################################################
31	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
32	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:7)
33	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:3)
34	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:4)
35	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:22)
36	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.utils.DatabaseChangeDecorator.update(DatabaseChangeDecorator.java:1)
37	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.DB.update(DB.java:3)
38	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.DB.update(DB.java:1)
39	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.Decks.flush(Decks.java:11)
40	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.Collection.save(Collection.java:8)
41	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.Collection.save(Collection.java:1)
42	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.sync.Syncer.sync(Syncer.java:4)
43	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doInBackgroundSync(Connection.java:16)
44	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doOneInBackground(Connection.java:2)
45	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doInBackground(Connection.java:4)
46	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doInBackground(Connection.java:1)
47	09-23 23:38:10.270 E/AnkiDroid( 1694): at android.os.AsyncTask$2.call(AsyncTask.java:304)
48	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.util.concurrent.FutureTask.run(FutureTask.java:237)
49	09-23 23:38:10.270 E/AnkiDroid( 1694): at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
50	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
51	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
52	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.lang.Thread.run(Thread.java:762)
53	09-23 23:38:10.270 E/AnkiDroid( 1694):
54	09-23 23:38:10.270 E/AnkiDroid( 1694): android.database.sqlite.SQLiteReadOnlyDatabaseException: attempt to write a readonly database (code 1032)
55	09-23 23:38:10.270 E/AnkiDroid( 1694): #################################################################
56	09-23 23:38:10.270 E/AnkiDroid( 1694): Error Code : 1032 (SQLITE_READONLY_DBMOVED)
57	09-23 23:38:10.270 E/AnkiDroid( 1694): Caused By : Database or Journal file have been removed.
58	09-23 23:38:10.270 E/AnkiDroid( 1694): (attempt to write a readonly database (code 1032))
59	09-23 23:38:10.270 E/AnkiDroid( 1694): #################################################################
60	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
61	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:7)
62	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:3)
63	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:4)
64	09-23 23:38:10.270 E/AnkiDroid( 1694): at io.requery.android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:22)
65	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.utils.DatabaseChangeDecorator.update(DatabaseChangeDecorator.java:1)
66	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.DB.update(DB.java:3)
67	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.DB.update(DB.java:1)
68	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.Decks.flush(Decks.java:11)
69	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.Collection.save(Collection.java:8)
70	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.Collection.save(Collection.java:1)
71	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.libanki.sync.Syncer.sync(Syncer.java:4)
72	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doInBackgroundSync(Connection.java:16)
73	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doOneInBackground(Connection.java:2)
74	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doInBackground(Connection.java:4)
75	09-23 23:38:10.270 E/AnkiDroid( 1694): at com.ichi2.async.Connection.doInBackground(Connection.java:1)
76	09-23 23:38:10.270 E/AnkiDroid( 1694): at android.os.AsyncTask$2.call(AsyncTask.java:304)
77	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.util.concurrent.FutureTask.run(FutureTask.java:237)
78	09-23 23:38:10.270 E/AnkiDroid( 1694): at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
79	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
80	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
81	09-23 23:38:10.270 E/AnkiDroid( 1694): at java.lang.Thread.run(Thread.java:762)
```

**Comment by david-allison-1** [comment]
This has been happening for some time - sync related and recoverable from what I understand.

Quite a lot of errors of this sort in ACRA

**Comment by david-allison-1** [comment]
Line of failure seems to be: 

https://github.com/ankidroid/Anki-Android/blob/63f82fa2a58ab7f04260dc0eaabe6702ed4155fe/AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java#L263-L273

**Comment by dae** [comment]
Closed by #14171.
